woocommerce/woocommerce
# Product Collection: Don't show publicly non-queryable taxonomies

### Changes proposed in this Pull Request:





> [!NOTE] 
> Same change was made in [this Gutenberg PR](https://github.com/WordPress/gutenberg/pull/59458)

This PR excludes taxonomies with `publicly_queryable` set to `false` from the Filter panel in the Product Collection block.

**Why?**
The behavior of the Product Collection block must be consistent in the editor and front end. `publicly_queryable` defines whether it is queryable on the front end, and if the value is `false`, the query results will not be displayed. Therefore, the editor should display taxonomies where `publicly_queryable` is `false` in the first place.

Closes # .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Add the following code to the Theme Functions (functions.php) file. This code adds queryable and non-queryable taxonomies to Product and Page, respectively.
```php
function add_taxonomies() {
	register_taxonomy( 'queryable_product_taxonomy', array( 'product' ), array(
		'public'             => true,
		'publicly_queryable' => true,
		'show_in_rest' 		   => true,
		'label'              => 'Queryable Product Taxonomy',
	) );
	register_taxonomy( 'non_queryable_product_taxonomy', array( 'product' ), array(
		'public'             => true,
		'publicly_queryable' => false,
		'show_in_rest' 		   => true,
		'label'              => 'Non Queryable Product Taxonomy',
	) );
	register_taxonomy( 'queryable_page_taxonomy', array( 'page' ), array(
		'public'             => true,
		'publicly_queryable' => true,
		'show_in_rest' 		   => true,
		'label'              => 'Queryable Page Taxonomy',
	) );
	register_taxonomy( 'non_queryable_page_taxonomy', array( 'page' ), array(
		'public'             => true,
		'publicly_queryable' => false,
		'show_in_rest' 		   => true,
		'label'              => 'Non Queryable Page Taxonomy',
	) );
}
add_action( 'init', 'add_taxonomies' );
```

2. Insert a Product Collection block in the Post Editor.
3. Add Taxonomies filter in Inspector controls . Taxonomies where `publicly_queryable` is `false` should not be displayed.

| Before | After |
|----------|----------|
| <img width="283" alt="image" src="https://github.com/woocommerce/woocommerce/assets/16707866/eff4fc57-c818-42da-94e3-cefe8d6b9dd1">  |   <img width="283" alt="image" src="https://github.com/woocommerce/woocommerce/assets/16707866/0b494149-670e-4d98-957b-988870cc5e57"> |



### Changelog entry




<details>

#### Significance




#### Type




#### Message 
Product Collection: Don't show publicly non-queryable taxonomies

#### Comment 

</details>
